miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/component/TimeZoneCorrector.java

```diff
@@ -0,0 +1,35 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller.component;
+
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.function.Consumer;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TimeZoneCorrector {
+
+  private final TimeZone dbZone;
+  private final TimeZone uiZone;
+
+  @Autowired
+  public TimeZoneCorrector(@Value("${miso.timeCorrection.uiZone:UTC}") String uiZone,
+      @Value("${miso.timeCorrection.dbZone:UTC}") String dbZone) {
+    this.dbZone = TimeZone.getTimeZone(dbZone);
+    this.uiZone = TimeZone.getTimeZone(uiZone);
+  }
+
+  public void toDbTime(Date date, Consumer<Date> setter) {
+    if (date == null) {
+      return;
+    }
+    long time = date.getTime();
+    int fromOffset = uiZone.getOffset(time);
+    int toOffset = dbZone.getOffset(time);
+    long shifted = time - fromOffset + toOffset;
```

Comment:
This logic might not work around leap days, but I'm not sure a better way without the new `java.time` stuff. So, :man_shrugging:

Reply:
It's only operating on the long value (ms since epoch), so leap days don't really have any effect